cmd/kubelet/app/server.go

```diff
@@ -626,6 +629,18 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 		runAuthenticatorCAReload(ctx.Done())
 	}
 
+	err = kubelet.PreInitRuntimeService(&s.KubeletConfiguration, kubeDeps)
+	if err != nil {
+		return err
+	}
+
+	// Get cgroup driver setting from CRI
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletCgroupDriverFromCRI) {
+		if err := getCgroupDriverFromCRI(ctx, s, kubeDeps); err != nil {
+			return err
+		}
+	}
```

Comment:
Do we do this:
- per new connection to the socket?
- per kubelet startup?

I'm thinking that if someone restarts the process at the other end of the socket, we should re-query this setting.

Reply:
This happens only at kubelet startup.

Technically we could query it at CRI reconnection but I wouldn't go there.

1. The cgroup driver is really not something that's expected to be changed without taking the node down. Changing it correctly requires restarting all pods, also the static ones. System reboot is a sage method.
2. I don't know what all would break in kubelet (internally) if we changed the cgroup-driver in-flight

We could re-query the cgroup driver at re-connection and exit if the cgroup driver setting has changed. But I'm not sure if that's desired. Thoughts @SergeyKanzhelev @mrunalp @haircommander

Reply:
I think our efforts would be better spent making it very clear a change in the cgroup driver in the CRI requires a node reboot